Equilobe.DailyReport.SL/AtlassianWorklogDataService.cs

```diff
@@ -0,0 +1,123 @@
+﻿using Equilobe.DailyReport.DAL;
+using Equilobe.DailyReport.Models.Dashboard;
+using Equilobe.DailyReport.Models.Interfaces;
+using Equilobe.DailyReport.Models.Storage;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Equilobe.DailyReport.SL
+{
+    public class AtlassianWorklogDataService : IAtlassianWorklogDataService
+    {
+        #region IAtlassianWorklogDataService implementation
```

Comment:
Why do you need to use this dictionaries? I think we are over complicating this. We can just get the worklogs by user.

Reply:
I already told you. For efficiency. It's not so complicated at all. I store the worklogs based on the user id so that I will not make a new call for each user id. It's a nice optimization that I use whenever I can - instead of making N calls for one element, I make one call with a list and check for containing, instead of equality and return a dictionary, for easy manipulation.